### PR TITLE
Hackathon module

### PR DESCRIPTION
**Pull request by arnaudroystatcan**
Hackathon module

Fix: #117

### CODE REVIEW

File: src/Modules/StatCan.OrchardCore.VueForms/wwwroot/Scripts/vue-forms.js [hunk 1]
```diff
@@ -120,8 +120,9 @@ function initForm(app) {
         var _this = this;
 
         e.preventDefault(); // cleanup any error / server success message
-      
-        // keep a reference to the VeeValidate observer
+
+        Object.assign(this.$data, this.$options.data.apply(this)); // keep a reference to the VeeValidate observer
```
Comment (jptissot) [inline src/Modules/StatCan.OrchardCore.VueForms/wwwroot/Scripts/vue-forms.js:124]: This should probably not be here 🤔

File: src/Themes/HackathonTheme/ResourceManifest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+namespace StatCan.Themes.HackathonTheme
+{
+    public class ResourceManifest : IResourceManifestProvider
+    {
+        public void BuildManifests(IResourceManifestBuilder builder)
+        {
+            var manifest = builder.Add();
+
+            manifest
+                .DefineStyle("PortalTheme-stc-bootstrap")
```
Comment (jptissot) [inline src/Themes/HackathonTheme/ResourceManifest.cs:12]: This theme should be revesited, this should be changed to the hackathon theme

File: src/Themes/HackathonTheme/ResourceManifest.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                .DefineScript("popper")
+                .SetUrl("~/PortalTheme/Scripts/popper.min.js", "~/PortalTheme/Scripts/popper.js")
+                .SetCdn("https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.min.js", "https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.js")
+                .SetCdnIntegrity("sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN", "sha384-cpSm/ilDFOWiMuF2bj03ZzJinb48NO9IGCXcYDtUzdP5y64Ober65chnoOj1XFoA")
+                .SetVersion("1.16.1");
+
+            manifest
+                .DefineScript("bootstrap")
+                .SetDependencies("jQuery", "popper")
+                .SetUrl("~/PortalTheme/Scripts/bootstrap.min.js", "~/PortalTheme/Scripts/bootstrap.js")
```
Comment (jptissot) [inline src/Themes/HackathonTheme/ResourceManifest.cs:44]: These links probably won't work

File: src/Themes/HackathonTheme/Views/Widget-TeamDashboardWidget.liquid [hunk 4]
```diff
@@ -0,0 +1,98 @@
+{% if User | is_in_role: "Hacker" %}
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/Widget-TeamDashboardWidget.liquid:1]: I think these files can stay in the module only

Comment (arnaudroystatcan) [reply src/Themes/HackathonTheme/Views/Widget-TeamDashboardWidget.liquid:1]: I forgot to remove them after the last merge, the good ones are in the module. I just pushed an update.

File: src/Modules/StatCan.OrchardCore.Hackathon/Services/IHackathonService.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        Task<bool> TeamExists(string teamContentItemId);
+        Task<bool> IsTeamFull(string teamContentItemId);
+        Task<ContentItem> CreateTeam(ModelStateDictionary modelState);
+        Task<ContentItem> JoinTeam(string teamContentItemId, ModelStateDictionary modelState);
+        Task<bool> LeaveTeam(ModelStateDictionary modelState);
+        Task<bool> MatchTeams();
+        string RootUrl(string hackathonSlug);
+        Task<bool> RunUpdateHandlers();
+    }
+}
```
Comment (jptissot) [inline src/Modules/StatCan.OrchardCore.Hackathon/Services/IHackathonService.cs:23]: EOF

**Comment by jptissot** [comment]
Let's start testing / fixing this PR so it can be merged soon. 

Things todo: 
- [ ] Code review
- [ ] Fix the forms (see the VueForms refactor #118)
- [ ] Add some e2e tests

**Review by jptissot — APPROVED** [review]

**Pull request merged**